src/Common.cpp

```diff
@@ -214,6 +214,18 @@ std::string NAS2D::trimWhitespace(std::string string)
 	return string.substr(first_non_space, last_non_space - first_non_space + 1);
 }
 
+bool NAS2D::startsWith(const std::string& string, const std::string& start)
+{
+	auto found_loc = string.find(start);
+	return found_loc != std::string::npos && found_loc == 0;
```

Comment:
This looks correct, though may suffer a bit in performance if `string` is unusually long, even if `start` was an immediate mismatch. In which case, the longer string could still be scanned in full, even though the function could terminate early.

Reply:
This could use a bit of polish. Specifically only check the first N characters of the input string via `std::string::substr`.